### ISSUE TITLE
Consider moving ANTLR out of the core module

### ISSUE THREAD

**Issue by ftomassetti** [Strumenta/kolasu#105]

**Comment by alessiostalla** [comment]
I've done some work in Tylasu for this: https://github.com/Strumenta/tylasu/commit/28b0aa8d2f99be95dfd9c48b75c3ca058ae323db and at the same time, I've tried to make Tylasu a little closer to how Kolasu is structured. So it could offer guidance on how to do the same thing in Kolasu.
I believe it's possible to make ANTLR optional with the current design (Node + Origin).

**Comment by ftomassetti** [comment]
Thank you, I started looking at your work on Tylasu and put together a draft PR: https://github.com/Strumenta/kolasu/pull/107 . I will further review your work and try to stay as aligned as possible

**Comment by ftomassetti** [comment]
I looked in the code in Tylasu and the most important change I noticed is in the metamodel: Node has no position, but the position is moved to Origin. Does it mean that the Node should not have positionOverride and positionOverride to move to Origin?

Note for the future: looking at the code in Tylasu I wonder if we should have a single metamodel definition stored in one place, instead of defining it Tylasu and Kolasu.
